### ISSUE TITLE
FScrawler issue while crawling through a remote host

### ISSUE THREAD

**Issue by usbhimani** [dadoonet/fscrawler#961]
There is an issue while crawling through files from remote host. Please see the code snippet below

**Job Settings**

```yml

---
name: "new_attachment"
server:
  hostname: "**************"
  port: 22
  username: "**********"
  password: "**************"
  protocol: "ssh"
fs:
  url: "/E:/foldertoindex"
  update_rate: "15m"
  excludes:
  - "*/~*"
  json_support: false
  filename_as_id: false
  add_filesize: true
  remove_deleted: true
  add_as_inner_object: false
  store_source: false
  index_content: true
  attributes_support: false
  raw_metadata: false
  xml_support: false
  index_folders: true
  lang_detect: false
  continue_on_error: false
  ocr:
    language: "eng"
    enabled: true
    pdf_strategy: "ocr_and_text"
  follow_symlinks: false
elasticsearch:
  nodes:
  - url: "http://*************:9200"
  bulk_size: 100
  flush_interval: "5s"
  byte_size: "10mb"

```

**Logs**

```
04:32:42,146 DEBUG [f.p.e.c.f.FsParserAbstract] Fs crawler thread [new_attachmen
t] is now running. Run #46...
04:32:42,146 DEBUG [f.p.e.c.f.c.s.FileAbstractorSSH] Opening SSH connection to _*username@hostname*_
04:32:42,783 DEBUG [f.p.e.c.f.c.s.FileAbstractorSSH] SSH connection successful
04:32:42,815 WARN  [f.p.e.c.f.FsParserAbstract] Error while crawling /E:/foldert
oindex: begin 0, end -1, length 17
04:32:42,815 WARN  [f.p.e.c.f.FsParserAbstract] Full stacktrace
java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 17
        at java.lang.String.checkBoundsBeginEnd(String.java:3319) ~[?:?]
        at java.lang.String.substring(String.java:1874) ~[?:?]
        at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.indexDirectory(Fs
ParserAbstract.java:541) ~[fscrawler-core-2.7-SNAPSHOT.jar:?]
        at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.run(FsParserAbstr
act.java:142) [fscrawler-core-2.7-SNAPSHOT.jar:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
04:32:42,819 DEBUG [f.p.e.c.f.FsParserAbstract] Fs crawler is going to sleep for
 15m
04:47:42,905 DEBUG [f.p.e.c.f.FsParserAbstract] Fs crawler is now waking up agai
n...
04:47:42,905 DEBUG [f.p.e.c.f.FsParserAbstract] Fs crawler thread [new_attachmen
t] is now running. Run #47...
04:47:42,905 DEBUG [f.p.e.c.f.c.s.FileAbstractorSSH] Opening SSH connection to D
IR\RGadge@NL0123EPC0921.dir.slb.com
04:47:43,589 DEBUG [f.p.e.c.f.c.s.FileAbstractorSSH] SSH connection successful
04:47:43,629 WARN  [f.p.e.c.f.FsParserAbstract] Error while crawling /E:/foldert
oindex: begin 0, end -1, length 17
04:47:43,629 WARN  [f.p.e.c.f.FsParserAbstract] Full stacktrace
java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 17
        at java.lang.String.checkBoundsBeginEnd(String.java:3319) ~[?:?]
        at java.lang.String.substring(String.java:1874) ~[?:?]
        at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.indexDirectory(Fs
ParserAbstract.java:541) ~[fscrawler-core-2.7-SNAPSHOT.jar:?]
        at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.run(FsParserAbstr
act.java:142) [fscrawler-core-2.7-SNAPSHOT.jar:?]
```

**Expected behavior**

The code should crawl through all the files on the remote server & create the index

**Versions:**
 - OS: Windows Server 2012 R2
 - FSCrawler version : **fscrawler-es7-2.7-SNAPSHOT**

**Comment by usbhimani** [comment]
Hi David,

Hope you are doing well!!
I just wanted to check on the issue. Can you please suggest by when will it be resolved?

Regards,
Umesh

**Comment by dadoonet** [comment]
@usbhimani Thanks for opening the issue. I did not have time yet to look at it. Sorry for this.
No idea on when I'll be able to look. 😞

**Comment by usbhimani** [comment]
@dadoonet : No worries, i can understand that. Is there any temporary workaround which I can use to proceed further?

**Comment by dadoonet** [comment]
I need to understand the issue to provide either a patch or a workaround.

**Comment by usbhimani** [comment]
Ok, do I need to provide more details?

**Comment by dadoonet** [comment]
Not yet. But I'll ask if needed. Thanks.

**Comment by dadoonet** [comment]
Are you running FSCrawler on a Linux machine by any chance?

**Comment by usbhimani** [comment]
We are using Windows VM

**Comment by dadoonet** [comment]
Thanks for confirming. I just merged #973 which should bring the fix. Let's wait for the next SNAPSHOT to be released in https://oss.sonatype.org/content/repositories/snapshots/fr/pilato/elasticsearch/crawler/fscrawler-es7/2.7-SNAPSHOT/

**Comment by usbhimani** [comment]
I'll check on this & let you know if there is any further issue.
